src/modules/history/HistoryWrapper.tsx

```diff
@@ -0,0 +1,287 @@
+import {
+  Box,
+  Button,
+  CircularProgress,
+  SvgIcon,
+  Typography,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { ConnectWalletPaper } from 'src/components/ConnectWalletPaper';
+import { ListWrapper } from 'src/components/lists/ListWrapper';
+import { SearchInput } from 'src/components/SearchInput';
+import { applyTxHistoryFilters, useTransactionHistory } from 'src/hooks/useTransactionHistory';
+import { useWeb3Context } from 'src/libs/hooks/useWeb3Context';
+
+import { downloadData, formatTransactionData, groupByDate } from './helpers';
+import { HistoryFilterMenu } from './HistoryFilterMenu';
+import { HistoryItemLoader } from './HistoryItemLoader';
+import { HistoryWrapperMobile } from './HistoryWrapperMobile';
+import TransactionRowItem from './TransactionRowItem';
+import { FilterOptions, TransactionHistoryItemUnion } from './types';
+
+export const HistoryWrapper = () => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [loadingDownload, setLoadingDownload] = useState(false);
+  const [filterQuery, setFilterQuery] = useState<FilterOptions[]>([]);
+  const [searchResetKey, setSearchResetKey] = useState(0);
+
+  const isFilterActive = searchQuery.length > 0 || filterQuery.length > 0;
+
+  const {
+    data: transactions,
+    isLoading,
+    fetchNextPage,
+    isFetchingNextPage,
+    fetchForDownload,
+  } = useTransactionHistory({ isFilterActive });
+
+  const handleJsonDownload = async () => {
+    setLoadingDownload(true);
+    const data = await fetchForDownload({ searchQuery, filterQuery });
+    const formattedData = formatTransactionData({ data, csv: false });
+    const jsonData = JSON.stringify(formattedData, null, 2);
+    downloadData('transactions.json', jsonData, 'application/json');
+    setLoadingDownload(false);
+  };
+
+  const handleCsvDownload = async () => {
+    setLoadingDownload(true);
+    const data: TransactionHistoryItemUnion[] = await fetchForDownload({
+      searchQuery,
+      filterQuery,
+    });
+    const formattedData = formatTransactionData({ data, csv: true });
+
+    // Getting all the unique headers
+    const headersSet = new Set<string>();
+    formattedData.forEach((transaction: TransactionHistoryItemUnion) => {
+      Object.keys(transaction).forEach((key) => headersSet.add(key));
+    });
+
+    const headers: string[] = Array.from(headersSet);
+    let csvContent = headers.join(',') + '\n';
+
+    formattedData.forEach((transaction: TransactionHistoryItemUnion) => {
+      const row: string[] = headers.map((header) => {
+        const value = transaction[header as keyof TransactionHistoryItemUnion];
+        if (typeof value === 'object') {
+          return JSON.stringify(value) ?? '';
+        }
+        return String(value) ?? '';
+      });
+      csvContent += row.join(',') + '\n';
+    });
+
+    downloadData('transactions.csv', csvContent, 'text/csv');
```

Comment:
Makes sense, it should be extremely rare since the only part that can fail is the `fetchForDownload` but I'll add it to a ticket of potential future changes